packages/vue-urql/src/useSubscription.ts

```diff
@@ -254,9 +254,12 @@ export function callUseSubscription<
   const extensions: Ref<Record<string, any> | undefined> = ref();
 
   const scanHandler = ref(handler);
-  const isPaused: Ref<boolean> = isRef(_args.pause)
-    ? _args.pause
-    : ref(!!_args.pause);
+  const isPaused: Ref<boolean> =
+    typeof _args.pause === 'function'
+      ? computed(_args.pause)
```

Comment:
Does `computed` give us a mutable ref that can still be modified after this conversion by `pause()` and `resume()`?

Reply:
@kitten no. Also computed will give additional overhead unnecessarily. Why don't we just do ref + watch?

Reply:
@kitten You're right, sorry for not being careful about that!

@negezor are you suggesting the following?
```
const isPaused: Ref<boolean> = ref(!!unref(args.pause));
watch(
  () => !!unref(args.pause),
  value => (isPaused.value = value)
);
```
Should we add the watcher to `stops` then?

Another possibility is to compute the value of `isPaused` from `args.pause`
```
const isPaused: Ref<boolean> = computed(() => !!unref(args.pause));
```
and have the latter updated by `pause` and `resume`:
```
pause() {
  args.pause = true;
},
resume() {
  args.pause = false;
},
```
Not sure which one is better...

Reply:
I pushed the ref + watch option. I guess the watcher should be added to `stops`, but I'm not sure yet.
Actually I do not understand why stops are being called (before unmounting) with `useClientHandle()` but not with `useClient()`...

Reply:
We can do a little optimization for watch, what do you think? It wouldn't require an unref in watch.
```ts
if (isRef(args.pause) || typeof args.pause === 'function') {
    watch(args.pause, value => (isPaused.value = value));
}
```

Reply:
You're right! What about adding it to the stops?

Reply:
It makes sense to add it to stops since it is used for `useClientHandle()`
https://github.com/urql-graphql/urql/blob/d95baf18d34e6337ecc9f7c9ce372028308c1b7b/packages/vue-urql/src/useClientHandle.ts#L130-L146

Reply:
@kitten maybe we can remove stops from the parameters and use [effectScope](https://vuejs.org/api/reactivity-advanced.html#effectscope)? This will eliminate the unnecessary need to store the stops array for most cases, which will have a positive impact on performance, as we generally do not use `useClientHandle()`.

Reply:
Thank you @negezor, I added the watchers to `stops`. Still, I do not understand: Why is it needed since all watchers are called synchronously? And why with useClientHandle specifically?